doc/install/build_install.rst

```diff
@@ -456,34 +455,38 @@ use ``ARB_PYTHON_LIB_PATH`` to specify the location where the Python module is t
 
     cmake .. -DARB_WITH_PYTHON=on -DARB_PYTHON_PATH_LIB=/custom/path
 
-If CMake is run in a `venv` or Conda environment, set ``CMAKE_INSTALL_PREFIX`` to the
-base path of the venv. The example below shows a workflow that creates a virtual
-environment, then installs Arbor inside the environment.
+.. note::
+    The location of libraries under a prefix in only guaranteed to be standard for Pythons global library location.
+    Therefore, correct installation of the Python package to any other location using ``CMAKE_INSTALL_PREFIX``,
+    such as user directory (e.g. `~/.local`), a Python or Conda virtual environment, may result in installation to a wrong path.
+    
+    ``python3 -m site --user-site`` (for user installations) or a path from ``python3 -c 'import site; print(site.getsitepackages())'``
+    (for virtual environment installation) can be used in combination with ``ARB_PYTHON_LIB_PATH``.
+
+    In addition, installation via ``pip`` or ``python setup.py`` is guaranteed to find the right path. Please refer to the
+    :ref:`Python installation instruction <in_python_custom>`.
+
 
 .. code-block:: bash
 
+    # A demonstration using ARB_PYTHON_LIB_PATH
+
     # Set up your venv.
     mkdir myenv
     cd myenv/
     python3 -m venv env
     source env/bin/activate
 
     # Install dependencies
-    pip install numpy
+    pip3 install numpy
 
     # Obtain arbor
     git clone --recursive git@github.com:arbor-sim/arbor.git
 
-    # Determine the prefix path for your installation
-    # method 1: set it explicitly
-    export pyprefix=$(pwd)/env
-    # method 2: query python directly by instpecting the output of sysconfig.
-    # Note that the ending of the form lib/python3.x/site-packages must be dropped.
-    # E.g if the output was /home/xxx/myenv/env/lib/python3.9/site-packages, we
-    # want to set pyprefix=/home/xxx/myenv/env/
-    python -c "import sysconfig; print(sysconfig.get_path('platlib'))"
+    # Get the user site-package directory
+    pyprefix=`python3 -c 'import site; print(site.getsitepackages()[0])'`
```

Comment:
So `import site; print(site.getsitepackages()[0])` is the new query? IIRC there was a problem with `virtualenv`s and the `site` module. Is this case covered? 

You also mention that only `pip` finds the correct location in all cases, but we're not querying pip here?

The 2 commands you mention provide different outputs for me:
```
> python -m site --user-site
/home/robin/.local/lib/python3.9/site-packages
```
```
python -c "import site; print(site.getsitepackages()[0])"
/home/robin/.pyenv/versions/vvv/lib/python3.9/site-packages
```

Reply:
Very confused, which is my I propose that Pythonic installation just uses pip, which should always work out as expected.

Indeed, those commands produce different outputs, because you may want to install the package differently. Either as a local user package `-m site --user-site`, or in the venv you're in `-c "import site; print(site.getsitepackages()[0])"`. At least, I assume those are the two main options that a Python user is interested in, because a system install into the default CMake prefix.

The point of this PR is to make clear that using CMake to install the Python package is going to be very unpythonic, and if that is what you want, you should really use pip. Does the documentation succeed in conveying that message?

Reply:
OK, I think it does. I googled "arbor python installation" and basically got `pip3 install arbor` which is good 👍 Then I went to the docs homepage and read through the "Get Arbor" text which also pointed me to the Python installation and `pip3 install arbor` or to build from source, where these paths seem to make their entry and the note reasonably explains that the path might be tricky to get right but the explanation of `-DARB_PYTHON_PATH_LIB` is there.

Reply:
:+1:

Of course, let me know if you find out a better way.